src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -243,6 +269,9 @@ Status CoreWorkerPlasmaStoreProvider::Wait(
       RAY_RETURN_NOT_OK(check_signals_());
     }
   }
+  if (ctx.CurrentTaskIsDirectCall() && ctx.ShouldReleaseResourcesOnBlockingCalls()) {
```

Comment:
How about we do the same thing as for the raylet path and call `HandleDirectCallTaskBlocked/Unblocked` directly from the node manager when processing/returning from the `Wait`? I think that's cleaner than doing separate IPCs from the client.

Reply:
The reason I did it this way is for blocking and unblocking to be consistently done in the new path. It's confusing when sometimes blocking is implicit and sometimes not. Performance wise, we are in the slow path for direct calls so an extra IPC can't hurt.

Reply:
Okay, sounds good.